### PR TITLE
Adding support for Select in QueryEnhancedRequest

### PR DESCRIPTION
**Pull request by shetsa-amzn**
Adding support for Select in QueryEnhancedRequest

## Motivation and Context
https://github.com/aws/aws-sdk-java-v2/issues/2083

## Modifications
Adding support for Select Attribute in QueryEnhancedRequest

https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html#DDB-Query-request-Select

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [x ] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [] My change requires a change to the Javadoc documentation
- [x] I have updated the Javadoc documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

**Comment by joviegas** [comment]
[ScanInput](https://github.com/aws/aws-sdk-java-v2/blob/81d0e84597487a6a55a392e343a6f45b36784172/services/dynamodb/src/main/resources/codegen-resources/dynamodb/service-2.json#L5521-L5544) also has the Select option , can we also support it for  ScanEnhancedRequest too ?

File: services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/functionaltests/document/BasicQueryTest.java [hunk 1]
```diff
@@ -238,6 +239,66 @@ public void queryAllRecordsDefaultSettings_withProjection() {
         assertThat(firstRecord.getNumber("value").intValue(), is(0));
     }
 
+    @Test
+    public void queryAllRecordsDefaultSettings_withSelect_specificAttributes() {
```
[the remark this answers is not in the document]
Comment (shetsa-amzn) [reply services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/functionaltests/document/BasicQueryTest.java:243]: Will do, Thanks.

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/QueryEnhancedRequest.java [hunk 2]
```diff
@@ -271,6 +301,19 @@ public Builder scanIndexForward(Boolean scanIndexForward) {
             return this;
         }
 
+        /**
+         * Determines the attributes to be returned in the result. See {@link Select} for examples and constraints.
+         * By default, all attributes are returned.
+         * @param select
+         * @return a builder of this type
+         */
+
```
[the remark this answers is not in the document]
Comment (shetsa-amzn) [reply services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/QueryEnhancedRequest.java]: Fixed.

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/QueryEnhancedRequest.java [hunk 3]
```diff
@@ -271,6 +301,19 @@ public Builder scanIndexForward(Boolean scanIndexForward) {
...
+        public Builder select(Select select) {
+            this.select = select;
+            return this;
+        }
+
+
```
[the remark this answers is not in the document]
Comment (shetsa-amzn) [reply services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/QueryEnhancedRequest.java]: Fixed

File: services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/QueryEnhancedRequest.java [hunk 4]
```diff
@@ -354,7 +397,7 @@ public Builder attributesToProject(Collection<String> attributesToProject) {
             }
             if (attributesToProject != null) {
                 addNestedAttributesToProject(new ArrayList<>(attributesToProject).stream()
-                        .map(NestedAttributeName::create).collect(Collectors.toList()));
+                                                                                 .map(NestedAttributeName::create).collect(Collectors.toList()));
```
[the remark this answers is not in the document]
Comment (shetsa-amzn) [reply services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/QueryEnhancedRequest.java]: Fixed

**Comment by shetsa-amzn** [comment]
> [ScanInput](https://github.com/aws/aws-sdk-java-v2/blob/81d0e84597487a6a55a392e343a6f45b36784172/services/dynamodb/src/main/resources/codegen-resources/dynamodb/service-2.json#L5521-L5544) also has the Select option , can we also support it for ScanEnhancedRequest too ?

As discussed will be addresses as we separate PR.

**Review by cenedhryn — APPROVED** [review]

**Review by cenedhryn — COMMENTED** [review]
The build is still failing with checkstyle errors?

**Review by dagnir — APPROVED** [review]

**Pull request merged**